src/AxelarGateway.sol

```diff
@@ -128,11 +180,12 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
             // If token address is no specified, it indicates a request to deploy one.
             bytes32 salt = keccak256(abi.encodePacked(symbol));
             tokenAddress = address(new BurnableMintableCappedERC20{ salt: salt }(name, symbol, decimals, cap));
+            _setTokenType(symbol, TokenType.InternalBurnableFrom);
         } else {
             // If token address is specified, ensure that there is a contact at the specified addressed.
-            require(tokenAddress.code.length != uint256(0), 'NOT_TOKEN');
+            require(tokenAddress.codehash != EMPTY_HASH, 'NOT_TOKEN');
```

Comment:
Cheaper to check `tokenAddress.code.length != 0`.

Reply:
You are tight, reading just bytesize was implemented in `0.8.0`